### PR TITLE
Polyhedron demo: New API for new features

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes

This PR holds an API breaking update of the Polyhedron demo. This allows to add new features, like:
- moving long operations to another thread
- having multiple viewers
- using scene transparency

### CODE REVIEW

**Comment by maxGimeno** [comment]
A first version of the new API can be found [here](https://cgal.geometryfactory.com/~mgimeno/Three_new_API/Three/index.html) .
After it has been reviewed and approved, Further work will be performed on this branch.

**Comment by lrineau** [comment]
Note that this branch already conflicts with master.

File: Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.cpp [hunk 1]
```diff
@@ -84,26 +92,49 @@ struct Scene_surface_mesh_item_priv{
...
+    item->edge_containers[1] = new Edge_container(VI::PROGRAM_NO_SELECTION,
+                                       true);
     has_feature_edges = false;
     invalidate_stats();
     vertices_displayed = true;
     edges_displayed = true;
     faces_displayed = true;
     all_primitives_displayed = false;
+    has_vcolors = false;
+    has_fcolors = false;
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.cpp]: The body of the two constructors are identical. It should be factorized into a helper method of the class.

**Comment by maxGimeno** [comment]
Doc updated

File: Polyhedron/demo/Polyhedron/Scene_item.cpp [hunk 2]
```diff
@@ -181,7 +181,7 @@ void CGAL::Three::Scene_item::selection_changed(bool b)
     setSelected(true);
     d->was_selected = true;
     redraw();
-    QTimer::singleShot(500, [this]{setSelected(false); redraw();});
+    QTimer::singleShot(500, this, SLOT(unSelect()));
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Scene_item.cpp]: Alternatively, you could have used:
```C++
QTimer::singleShot(500, this, [this]{setSelected(false); redraw();});
```
with three arguments.

See the doc: http://doc.qt.io/qt-5/qtimer.html#singleShot-2

**Comment by maxGimeno** [comment]
Oh, thanks for the indication, that will be far better

2017-11-30 18:19 GMT+01:00 Laurent Rineau <notifications@github.com>:

**Comment by lrineau** [comment]
This change is [Reviewable](https://reviewable.io/reviews/cgal/cgal/2501)

**Comment by lrineau** [comment]
That reading might be interesting.

**Comment by lrineau** [comment]
http://gameprogrammingpatterns.com/dirty-flag.html

**Comment by sloriot** [comment]
We're closing this PR as we won't have time to finish it. Transparency and multiple viewers were added in another PR.

**Pull request closed without merging**